metadataStore/commands.py

```diff
@@ -318,23 +287,79 @@ def find_event_descriptor(_id):
     _id: bson.ObjectId
 
     """
+    connect(db=database, host=host, port=port)
     event_descriptor_list = list()
     connect(db=database, host=host, port=port)
-    for event_descriptor in EventDescriptor.objects(id=_id).order_by('-_id'):
+    for event_descriptor in EventDescriptor.objects(begin_run_event=begin_run_event.id).order_by('-_id'):
         event_descriptor = __replace_descriptor_data_key_dots(event_descriptor,
                                                               direction='out')
         event_descriptor_list.append(event_descriptor)
     return event_descriptor_list
 
 
-def find_event(header):
-    # TODO: Other search parameters for events?
+def fetch_events(limit=1000, **kwargs):
+    """
+
+    Parameters
+    -----------
+    limit: int
+        number of events returned
+
+    Other Parameters
+    ----------------
+    time: dict
+        time of the event. dict keys must be start and end
+    descriptor: mongoengine.Document
+        event descriptor object
+    """
     connect(db=database, host=host, port=port)
-    event_list = list()
-    for event in Event.objects(header=header.id).order_by('-_id'):
-        event = __replace_event_data_key_dots(event, direction='out')
-        event_list.append(event)
-    return event_list
+    search_dict = dict()
+    try:
+        time_dict = kwargs.pop('time')
+        if not isinstance(time_dict, dict):
```

Comment:
Same logic as above with the `try` block.  I think you have enough if staments here that you will never get the `KeyError` raised.

Probably worth pulling this out into a helper function.